timemachine/cpp/src/fanout_summed_potential.cu

```diff
@@ -1,14 +1,22 @@
-#include "energy_accumulation.hpp"
 #include "fanout_summed_potential.hpp"
 #include "gpu_utils.cuh"
 #include "nonbonded_common.hpp"
+#include <cub/cub.cuh>
 #include <memory>
 
 namespace timemachine {
 
 FanoutSummedPotential::FanoutSummedPotential(
     const std::vector<std::shared_ptr<Potential>> potentials, const bool parallel)
-    : potentials_(potentials), parallel_(parallel), d_u_buffer_(potentials_.size()) {};
+    : potentials_(potentials), parallel_(parallel), d_u_buffer_(potentials_.size()), sum_storage_bytes_(0) {
+
+    gpuErrchk(
+        cub::DeviceReduce::Sum(nullptr, sum_storage_bytes_, d_u_buffer_.data, d_u_buffer_.data, potentials_.size()));
+
+    gpuErrchk(cudaMalloc(&d_sum_temp_storage_, sum_storage_bytes_));
+};
+
+FanoutSummedPotential::~FanoutSummedPotential() { gpuErrchk(cudaFree(d_sum_temp_storage_)); };
```

Comment:
can `DeviceBuffer` be used for cases like this where the lifetime of the buffer is the same as the lifetime of the object?